plotly/tools.py

```diff
@@ -1443,6 +1449,398 @@ class FigureFactory(object):
     """
 
     @staticmethod
+    def _unlabel_rgb(colors):
+        """
+        This function takes a list of two 'rgb(a, b, c)' color strings and
+        returns a list of the color tuples in tuple form without the 'rgb'
+        label. In particular, the output is a list of two tuples of the form
+        (a, b, c)
+        """
+        unlabelled_colors = []
+        for color in colors:
+            str_vals = ''
+            for index in range(len(color)):
+                try:
+                    float(color[index])
+                    str_vals = str_vals + color[index]
+                except ValueError:
+                    if (color[index] == ',') or (color[index] == '.'):
+                        str_vals = str_vals + color[index]
+
+            str_vals = str_vals + ','
+            numbers = []
+            str_num = ''
+            for char in str_vals:
+                if char != ',':
+                    str_num = str_num + char
+                else:
+                    numbers.append(float(str_num))
+                    str_num = ''
+            unlabelled_tuple = (numbers[0], numbers[1], numbers[2])
+            unlabelled_colors.append(unlabelled_tuple)
+
+        return unlabelled_colors
+
+    @staticmethod
+    def _find_intermediate_color(tuple1, tuple2, t):
```

Comment:
🐄 can we give `tuple1` and `tuple2` and `t` better variable names? Naming things is always more time-consuming than you want it to be, but it *is* helpful for code readability.